Source/WebKit/UIProcess/UserMediaPermissionRequestManagerProxy.cpp

```diff
@@ -307,8 +323,19 @@ void UserMediaPermissionRequestManagerProxy::finishGrantingRequest(UserMediaPerm
         return;
     }
 
-    protectedPage()->willStartCapture(request, [this, weakThis = WeakPtr { *this }, request = Ref { request }]() mutable {
-        if (!weakThis)
+    RefPtr page = m_page.get();
+    if (!page) {
+        denyRequest(request, UserMediaPermissionRequestProxy::UserMediaAccessDenialReason::OtherFailure, "No page."_s);
+        return;
+    }
+
+    page->willStartCapture(request, [this, weakThis = WeakPtr { *this }, request = Ref { request }]() mutable {
+        RefPtr protectedThis = weakThis.get();
+        if (!protectedThis)
```

Comment:
In theory, we should probably deny the request here as well since it owns a CompletionHandler.
Or maybe have UserMediaPermissionRequestProxy call its completion handler (if any) in its destructor as a catch them all handling.